crates/wdl-analysis/src/handlers/goto_definition.rs

```diff
@@ -563,8 +603,99 @@ fn resolve_access_expression(
+            if original_enum.namespace().is_some() {
+                continue;
+            };
+
+            // SAFETY: we know `lines` will return Some as we only reach here when
+            // `node.document` is fully parsed and in `ParsedState::Parse`
+            // state.
+            let imported_lines = node.parse_state().lines().unwrap();
+
+            let enum_node =
+                v1::EnumDefinition::cast(SyntaxNode::new_root(original_enum.node().clone()))
+                    .expect("should cast to enum definition");
+
+            if let Some(variant) = enum_node
+                .variants()
+                .find(|v| v.name().text() == variant_ident.text())
+            {
+                let variant_span = variant.name().span();
+                let span = Span::new(
+                    variant_span.start() + original_enum.offset(),
+                    variant_span.len(),
+                );
+                return Ok(Some(location_from_span(ns.source(), span, imported_lines)?));
+            }
+        }
+
+        // Check for enum definition in local document.
+        let enum_def = analysis_doc.enum_by_name(enum_ty.name()).ok_or_else(|| {
+            anyhow!(
+                "definition not found for enum `{name}`",
+                name = enum_ty.name()
+            )
+        })?;
+
+        let (uri, def_lines) = match enum_def.namespace() {
+            Some(ns_name) => {
+                // SAFETY: `namespace` returns `Some` only when enum was imported from a
+                // namespace that exists in the document.
+                let ns = analysis_doc.namespace(ns_name).unwrap();
+
+                // SAFETY: `ns.source` comes from a valid namespace entry which guarantees the
+                // document exists in the graph.
+                let imported_node = graph.get(graph.get_index(ns.source()).unwrap());
+
+                // SAFETY: we successfully got the document above, it's in
+                // `ParseState::Parsed` which always has a valid lines field.
+                let lines = imported_node.parse_state().lines().unwrap();
+                (ns.source().as_ref(), lines)
+            }
+            None => (document_uri, lines),
+        };
+
+        let enum_node = v1::EnumDefinition::cast(SyntaxNode::new_root(enum_def.node().clone()))
+            .expect("should cast to enum definition");
+
+        let Some(variant) = enum_node
+            .variants()
+            .find(|v| v.name().text() == variant_ident.text())
+        else {
+            return Ok(None);
+        };
+
+        let variant_span = variant.name().span();
+        let span = Span::new(variant_span.start() + enum_def.offset(), variant_span.len());
+        // Returns found enum variant definition location.
+        return Ok(Some(location_from_span(uri, span, def_lines)?));
+    }
+
+    if let Type::TypeNameRef(CustomType::Struct(_)) = target_type {
+        todo!("handle struct member access via `TypeNameRef`")
```

Comment:
Should this be a `todo` rather than returning `Ok(None)` for the goto definition request, as `MyStruct.member` isn't a valid expression in WDL?